vignettes/datatable-programming.Rmd

```diff
@@ -376,6 +376,19 @@ print(j)
 DT[, j, env = list(j = j)]
 ```
 
+### Common mistakes
+
+It is important to understand the difference between passing an object and a name that points to an object. See the verbose output of following examples.
+
+```{r obj_vs_objname}
+DT[, fun(Petal.Width), env = list(fun = mean), verbose=TRUE]
+DT[, fun(Petal.Width), env = list(fun = "mean"), verbose=TRUE]
+```
+
+User will usually want to substitute function name rather than function body.
+
+In case of any doubts on the `env` interface functioning user is advised to look at `verbose` output.
```

Comment:
```suggestion
In case of any doubts on the `env` interface functioning, set `verbose = TRUE` to inspect how expressions are resolved internally.
```